### PR TITLE
`rasa data convert` should be able to convert forms to `YAML` format

### LINKED ISSUE #6501: Convert forms with `rasa convert`
Part of #6475

**Description of Problem**:
`rasa convert` should also convert forms, but warn users that the `FormPolicy` is deprecated.

### PR DESCRIPTION
**Pull request by degiz**
Closes #6501 

**Proposed changes**:
- `form` is converted to `active_loop`
- users are warned that `FormPolicy` has to be activated

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/io.py [hunk 1]
```diff
@@ -293,6 +293,12 @@ def write_yaml(
     # no wrap lines
     dumper.width = YAML_LINE_MAX_WIDTH
 
+    # use `null` to represent `None`
```
Comment (federicotdn) [inline rasa/utils/io.py:296]: `None` isn't mapped to `null` already? :thinking:

File: rasa/core/training/story_writer/yaml_story_writer.py [hunk 2]
```diff
@@ -91,17 +88,6 @@ def process_story_step(self, story_step: StoryStep) -> Optional[OrderedDict]:
...
-        if self.story_contains_forms(story_step):
-            raise_warning(
-                f'Training data file contains a story "{story_step.block_name}" '
-                f"that has form(s) in it. This story cannot be converted automatically "
-                f"because of the new Rules system in Rasa Open Source "
-                f"version {LATEST_TRAINING_DATA_FORMAT_VERSION}. "
-                f"Please convert this story manually, it will be skipped now.",
-                docs=DOCS_URL_STORIES,
-            )
-            return None
```
Comment (federicotdn) [inline rasa/core/training/story_writer/yaml_story_writer.py:103]: With this removed, is the return type of this function still `Optional`?

**Review by federicotdn — APPROVED** [review]
Tested it locally and it's working great :+1: :+1:

File: rasa/utils/io.py (hunk 1 above)
Comment (degiz) [reply rasa/utils/io.py:296]: Somehow no, it wasn't! I've checked the sources, and there's a condition indeed when `null` is replaced by just empty space, but I couldn't quickly figure out what leads to it 😞

File: rasa/core/training/story_writer/yaml_story_writer.py (hunk 2 above)
Comment (degiz) [reply rasa/core/training/story_writer/yaml_story_writer.py:103]: No, it's not, good catch!

**Pull request merged**